google/cloud/aiplatform/featurestore/featurestore.py

```diff
@@ -557,3 +562,597 @@ def create_entity_type(
+                destination_feature_settings.append(destination_feature_setting)
+
+        else:
+            raise TypeError(
+                f"The 'feature_destination_fields' for each entity_type should be a dict, list or set, "
+                f"instead, got {type(feature_destination_fields)}."
+            )
+
+        return feature_ids, destination_feature_settings
+
+    def _validate_and_get_batch_read_feature_values_request(
+        self,
+        entity_type_ids: List[str],
+        destination: Union[
+            gca_io.BigQueryDestination,
+            gca_io.CsvDestination,
+            gca_io.TFRecordDestination,
+        ],
+        entity_type_destination_fields: Optional[
+            Dict[str, Union[Dict[str, str], List[str], Set[str]]]
+        ] = None,
+        read_instances: Optional[Union[gca_io.BigQuerySource, gca_io.CsvSource]] = None,
+        pass_through_fields: Optional[List[str]] = None,
+    ) -> gca_featurestore_service.BatchReadFeatureValuesRequest:
+        """Validates and gets batch_read_feature_values_request
+
+        Args:
+            entity_type_ids (List[str]):
+                Required. ID of the EntityType to select batch serving Features. The
+                EntityType id is the specified during EntityType creation.
+            destination (Union[gca_io.BigQueryDestination, gca_io.CsvDestination, gca_io.TFRecordDestination]):
+                Required. BigQuery destination, Csv destination or TFRecord destination.
+            entity_type_destination_fields (Dict[str, Union[Dict[str, str], List[str], Set[str]]]):
+                Optional. User defined dictionary to map ID of the EntityType's Features
+                to the batch serving destination field name.
+
+                Specify the features to be batch served in each entityType, and their destination field name.
+                If the features are not specified, all features will be batch served.
+                If the destination field name is not specified, Feature ID will be used as destination field name.
+
+                Example:
+
+                     - In case all features will be batch served and using Feature ID as destination field name:
+
+                         entity_type_ids = ['my_entity_type_id_1', 'my_entity_type_id_2', 'my_entity_type_id_3']
+
+                         entity_type_destination_fields = {}
+                         or
+                         entity_type_destination_fields = {
+                            'my_entity_type_id_1': {},
+                            'my_entity_type_id_2': [],
+                            'my_entity_type_id_3': None,
+                         }
+
+                     - In case selected features will be batch served and using Feature ID as destination field name:
+
+                         entity_type_ids = ['my_entity_type_id_1', 'my_entity_type_id_2', 'my_entity_type_id_3']
+
+                         feature_source_fields = {
```

Comment:
Do you mean `entity_type_destination_fields`?

Reply:
fixed the docstring